examples/bank/bank.go

```diff
@@ -0,0 +1,214 @@
+				return err
+			}
+			fromValue = readValue
+		}
+
+		// Ensure there is enough cash.
+		if fromValue < amount {
+			return nil
+		}
+
+		// Read to value.
+		call = client.Get(proto.Key(to))
+		gr = call.Reply.(*proto.GetResponse)
+		if err := txn.Run(call); err != nil {
+			return err
+		}
+
+		var toValue int64
+		if gr.Value != nil && gr.Value.Bytes != nil {
+			readValue, err := strconv.ParseInt(string(gr.Value.Bytes), 10, 64)
+			if err != nil {
+				return err
+			}
+			toValue = readValue
+		}
+		// Update both accounts.
+		txn.Prepare(client.Put(proto.Key(from), []byte(fmt.Sprintf("%d", fromValue-amount))))
+		txn.Prepare(client.Put(proto.Key(to), []byte(fmt.Sprintf("%d", toValue+amount))))
+		return nil
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	bank.approxNumTransactions++
+	return true
+}
+
+// Read the balances in all the accounts and return them.
+func (bank *Bank) readAllAccounts() []int64 {
+	balances := make([]int64, bank.numAccounts)
+	// Key Value Client initialization.
+	sender, err := client.NewHTTPSender(bank.servingAddr, testutils.NewTestBaseContext())
+	if err != nil {
+		log.Fatal(err)
+	}
+	kvClient := client.NewKV(nil, sender)
+	kvClient.User = storage.UserRoot
+	txnOpts := &client.TransactionOptions{Name: "Reading all balances"}
+	err = kvClient.RunTransaction(txnOpts, func(txn *client.Txn) error {
+		for i := 0; i < bank.numAccounts; i++ {
+			call := client.Get(proto.Key(makeAccountID(i)))
+			gr := call.Reply.(*proto.GetResponse)
+			if err := txn.Run(call); err != nil {
+				return err
+			}
+
+			if gr.Value != nil && gr.Value.Bytes != nil {
+				balances[i], err = strconv.ParseInt(string(gr.Value.Bytes), 10, 64)
+				if err != nil {
+					return err
+				}
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	return balances
+}
+
+// continuouslyTransferMoney() keeps moving random amounts between
+// random accounts.
+func (bank *Bank) continuousMoneyTransfer() {
+	// Key Value Client initialization.
+	sender, err := client.NewHTTPSender(bank.servingAddr, testutils.NewTestBaseContext())
+	if err != nil {
+		log.Fatal(err)
+	}
+	kvClient := client.NewKV(nil, sender)
+	kvClient.User = storage.UserRoot
+
+	for {
+		from := makeAccountID(rand.Intn(bank.numAccounts))
+		to := makeAccountID(rand.Intn(bank.numAccounts))
+		exchange := rand.Int63n(100)
+		bank.moveMoney(kvClient, from, to, exchange)
+		// time.Sleep(time.Duration(rand.Int63n(10)) * time.Millisecond)
+	}
+}
+
+// Initialize all the bank balances
+func initBankAccounts(servingAddr string, numAccounts int, cash int64) {
+	// Key Value Client initialization.
+	sender, err := client.NewHTTPSender(servingAddr, testutils.NewTestBaseContext())
+	if err != nil {
+		log.Fatal(err)
+	}
+	kvClient := client.NewKV(nil, sender)
+	kvClient.User = storage.UserRoot
+
+	putOpts := client.TransactionOptions{Name: "Initialize bank"}
+	for i := 0; i < numAccounts; i++ {
+		err = kvClient.RunTransaction(&putOpts, func(txn *client.Txn) error {
```

Comment:
This initialization does not need to be performed in a transaction. It would be more efficient to not use one here. And you'd be better off run the puts in a batch. See my other comment about sending multiple calls to `Run`.